pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+			SecretType: "PASSWORD",
+			Comment:    comment,
+			Username:   "bar",
+		})
+
+		unsetRequest := sdk.NewAlterSecretRequest(id).
+			WithUnset(
+				*sdk.NewSecretUnsetRequest().
+					WithComment(true),
+			)
+		err = client.Secrets.Alter(ctx, unsetRequest)
+		require.NoError(t, err)
+
+		details, err = client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, details.Comment, sql.NullString{String: "", Valid: false})
+	})
+
+	t.Run("Alter: SecretWithGenericString", func(t *testing.T) {
+		comment := random.Comment()
+		_, id := createSecretWithGenericString(t, "foo", nil)
+		setRequest := sdk.NewAlterSecretRequest(id).
+			WithSet(
+				*sdk.NewSecretSetRequest().
+					WithComment(comment).
+					WithSetForGenericString(
+						*sdk.NewSetForGenericStringRequest().
+							WithSecretString("bar"),
+					),
+			)
+		err := client.Secrets.Alter(ctx, setRequest)
+		require.NoError(t, err)
+
+		unsetRequest := sdk.NewAlterSecretRequest(id).
+			WithUnset(
+				*sdk.NewSecretUnsetRequest().
+					WithComment(true),
+			)
+		err = client.Secrets.Alter(ctx, unsetRequest)
+		require.NoError(t, err)
+
+		details, err := client.Secrets.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assert.Equal(t, details.Comment, sql.NullString{String: "", Valid: false})
+	})
+
+	t.Run("Drop", func(t *testing.T) {
+		_, id := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{{Scope: "foo"}}, nil)
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NotNil(t, secret)
+		require.NoError(t, err)
+
+		err = client.Secrets.Drop(ctx, sdk.NewDropSecretRequest(id))
+		require.NoError(t, err)
+
+		secret, err = client.Secrets.ShowByID(ctx, id)
+		require.Nil(t, secret)
+		require.Error(t, err)
+	})
+
+	t.Run("Drop: non-existing", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+
+		err := client.Secrets.Drop(ctx, sdk.NewDropSecretRequest(id))
+		assert.ErrorIs(t, err, sdk.ErrObjectNotExistOrAuthorized)
+	})
+
+	t.Run("Show", func(t *testing.T) {
+		secretOAuthClientCredentials, _ := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{{Scope: "foo"}}, nil)
+		secretOAuthAuthorizationCode, _ := createSecretWithOAuthAuthorizationCodeFlow(t, integrationId, "foo", refreshTokenExpiryTime, nil)
+		secretBasicAuthentication, _ := createSecretWithBasicAuthentication(t, "foo", "bar", nil)
+		secretGenericString, _ := createSecretWithGenericString(t, "foo", nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest())
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secretOAuthClientCredentials)
+		require.Contains(t, returnedSecrets, *secretOAuthAuthorizationCode)
+		require.Contains(t, returnedSecrets, *secretBasicAuthentication)
+		require.Contains(t, returnedSecrets, *secretGenericString)
+	})
+
+	t.Run("Show: SecretWithOAuthClientCredentialsFlow with Like", func(t *testing.T) {
```

Comment:
It's cool you've provided all of these cases, but I think that one case for each filtering option is enough (with an arbitrary type).